packages/ilios-common/addon/components/course/rollover.js

```diff
@@ -31,27 +30,44 @@ export default class CourseRolloverComponent extends Component {
     for (let i = 0; i < 6; i++) {
       this.years.push(currentYear + i);
     }
+    this.title = this.args.course.title;
   }
 
-  load = restartableTask(async (event, [course]) => {
-    if (!course) {
-      return;
-    }
-    this.title = course.title;
-    const school = course.belongsTo('school').id();
-    this.academicYearCrossesCalendarYearBoundaries = await this.iliosConfig.itemFromConfig(
-      'academicYearCrossesCalendarYearBoundaries',
+  @cached
+  get academicYearCrossesCalendarYearBoundariesData() {
+    return new TrackedAsyncData(
+      this.iliosConfig.itemFromConfig('academicYearCrossesCalendarYearBoundaries'),
+    );
+  }
+
+  get academicYearCrossesCalendarYearBoundaries() {
+    return this.academicYearCrossesCalendarYearBoundariesData.isResolved
+      ? this.academicYearCrossesCalendarYearBoundariesData.value
+      : false;
+  }
+
+  @cached
+  get allCoursesData() {
+    const school = this.args.course.belongsTo('school').id();
+    return new TrackedAsyncData(
+      this.store.query('course', {
+        filters: {
+          school,
+        },
+      }),
     );
-    this.allCourses = await this.store.query('course', {
-      filters: {
-        school,
-      },
-    });
+  }
+
+  get allCourses() {
+    return this.allCoursesData.isResolved ? this.allCoursesData.value : null;
+  }
 
-    // set selectedYear to next valid year (or current if none found)
+  @action
+  async loadValidYear() {
+    await this.allCourses;
```

Comment:
This value can't be awaited. It will always be `null` and resolve instantly. I think you'll need to refactor this out into getters anyway, based on using the element modifier to load this, but worth nothing this troublespot I've hit before. The `TrackedAsyncData` pattern is nice, but it has this gotcha built in.

Reply:
Good point. Removed.